zcash_client_sqlite/src/wallet/init/migrations/transparent_gap_limit_handling.rs

```diff
@@ -60,6 +60,73 @@ impl<P, C, R> schemerz::Migration<Uuid> for Migration<P, C, R> {
     }
 }
 
+// For each account, ensure that all diversifier indexes prior to that for the default
+// address have corresponding cached transparent addresses.
+#[cfg(feature = "transparent-inputs")]
+pub(super) fn insert_initial_transparent_addrs<P: consensus::Parameters>(
+    conn: &rusqlite::Transaction,
+    params: &P,
+) -> Result<(), WalletMigrationError> {
+    let mut min_addr_diversifiers = conn.prepare(
+        r#"
+        SELECT accounts.id AS account_id,
+               MIN(addresses.transparent_child_index) AS transparent_child_index,
+               MIN(addresses.diversifier_index_be) AS diversifier_index_be
+        FROM accounts
+        LEFT OUTER JOIN addresses
+            ON accounts.id = addresses.account_id
+            AND addresses.key_scope = :key_scope_external
+        GROUP BY accounts.id
+        "#,
+    )?;
+
+    let mut min_addr_rows = min_addr_diversifiers.query(named_params![
+        ":key_scope_external": KeyScope::EXTERNAL.encode()
+    ])?;
+    while let Some(row) = min_addr_rows.next()? {
+        let account_id = AccountRef(row.get::<_, u32>("account_id")?);
+
+        let min_transparent_idx = row
+            .get::<_, Option<u32>>("transparent_child_index")?
+            .map(|i| {
+                NonHardenedChildIndex::from_index(i).ok_or(WalletMigrationError::CorruptedData(
+                    format!("{} is not a valid transparent child index.", i),
+                ))
+            })
+            .transpose()?;
+
+        let min_diversifier_idx = row
+            .get::<_, Option<Vec<u8>>>("diversifier_index_be")?
+            .map(|b| decode_diversifier_index_be(&b[..]))
+            .transpose()?
+            .and_then(|di| NonHardenedChildIndex::try_from(di).ok());
+
+        // Ensure that there is an address for each possible external address index prior to the
+        // default UA for the account. If the default address has a diversifier index greater than
+        // the gap limit, we generate transparent addresses up to that index but not beyond.
+        let start = NonHardenedChildIndex::const_from_index(0);
+        let end = std::cmp::min(
+            min_transparent_idx
+                .or(min_diversifier_idx)
+                .unwrap_or(NonHardenedChildIndex::const_from_index(1)),
```

Comment:
Document that we only generate 1 in this case (instead of gap limit) because this is in essence inserting the "legacy default t-addr" that we previously were manually inserting into the output of `get_transparent_receivers`. (It also means that we have an entry at index 0, and thus when an address is actually generated we create a gap that the gap-handling logic will handle.)